pybossa/view/fileproxy.py

```diff
@@ -93,11 +107,53 @@ def encrypted_file(store, bucket, project_id, path):
     cipher = AESWithGCM(secret)
     decrypted = cipher.decrypt(content)
 
-    ## respond
     response = Response(decrypted, content_type=_key.content_type)
-
     response.headers.add('Content-Encoding', _key.content_encoding)
     response.headers.add('Content-Disposition', _key.content_disposition)
-    response.headers.add('Cache-Control', 'no-store')
-    response.headers.add('Pragma', 'no-cache')
     return response
+
+
+@blueprint.route('/hdfs/<string:cluster>/<int:project_id>/<path:path>')
+@no_cache
+@login_required
+def hdfs_file(project_id, cluster, path):
+    signature = request.args.get('task-signature')
+    if not signature:
+        raise Forbidden('No signature')
+
+    project = get_project_data(project_id)
+    timeout = project['info'].get('timeout', ContributionsGuard.STAMP_TTL)
+    payload = signer.loads(signature, max_age=timeout)
+    task_id = payload['task_id']
+    check_allowed(current_user.id, task_id, project, request.path)
+
+    client = HDFSKerberos(**current_app.config['HDFS_CONFIG'][cluster])
+    try:
+        content = client.get('/{}'.format(path))
+        project_encryption = project['info'].get('ext_config', {}).get('encryption', {})
+        if project_encryption:
+            secret = get_secret_from_vault(project_encryption)
```

Comment:
So in this case, each project will have it's own secret for decrypting (which is different from the bcos files).  What is that vault?